src/sql_generator.cpp

```diff
@@ -327,9 +327,15 @@ void truncate_table(duckdb::Connection &con, const table_def &table,
   const std::string absolute_table_name = table.to_escaped_string();
   std::ostringstream sql;
 
-  sql << "UPDATE " << absolute_table_name << " SET "
-      << KeywordHelper::WriteQuoted(deleted_column, '"') << " = true WHERE "
-      << KeywordHelper::WriteQuoted(synced_column, '"')
+  if (deleted_column.empty()) {
+    // hard delete
+    sql << "DELETE FROM " << absolute_table_name;
```

Comment:
I noticed we are using `KeywordHelper::WriteQuoted` for column names. Should the table names also be quoted? Or are they already quoted before passing to this function?

Reply:
The individual components of the fully qualified table name are quoted in [table_def::to_escaped_string()](https://github.com/MotherDuck-Open-Source/motherduck-fivetran-connector/blob/a2e66b75b4fae8b6b8613cdae4954749115fc11d/src/sql_generator.cpp#L11).